src/openloco/window.cpp

```diff
@@ -117,6 +117,40 @@ namespace openloco::ui
             viewports[1]->render(dpi);
     }
 
+    // 0x0045FCE6
+    // Input:
+    // regs.ax:  x
+    // regs.bx:  y
+    // regs.bp:  z
+    // Output:
+    // {x: regs.ax, y: regs.bx}
+    // Note: function seems to be similar to screen_get_map_xy_with_z in openrct2
+    map::map_pos sub_45FCE6(int16_t x, int16_t y, int16_t z)
+    {
+        window* w = WindowManager::findAt(x, y);
+        if (w != nullptr)
+        {
+            viewport* vp = w->viewports[0];
+            if (vp != nullptr && x >= vp->x && x < vp->x + vp->width && y >= vp->y && y < vp->y + vp->height)
```

Comment:
This is going to be used quite a bit in different functions. We should make a method on viewport for a `containsUi`.

Reply:
I added this function.